src/accelerate/tracking.py

```diff
@@ -54,6 +54,14 @@ class GeneralTracker(object, metaclass=ABCMeta):
     A base Tracker class to be used for all logging integration implementations.
     """
 
+    @abstractproperty
+    def requires_logging_directory(self):
+        """
+        Whether the logger requires a directory to store their logs. Should either return `True` or `False`. The
+        default behavior is `False`
+        """
+        pass
```

Comment:
Why not use a class attribute directly? We could set a default then (I don't think this actually defaults to False).

Reply:
This won't right. (This is also evidence of me going back and forth with it). Having it like this IMO forces implementations to think about how they want the behavior to go, and have it explicit in the source code for users interested in the code to see. 

For example, forcing it to be an abstract property/function, W&B could eventually make it a conditional where if it's in `offline_mode` (by checking the env var), then it would be True, otherwise False.

But, I'm not against just making it a normal class attr directly

Reply:
In that case, you should remove the default behavior mention in the docstring, as there is no default :-)